src/environ/secrets.py

```diff
@@ -113,13 +135,80 @@ def _get(self, environ, metadata, prefix, name):
             var = "_".join((prefix[1:] + (name,)))
         try:
             log.debug("looking for '%s' in section '%s'." % (var, section))
-            return _SecretStr(self._cfg.get(section, var))
+            val = self._cfg.get(section, var)
+            return _SecretStr(val)
         except NoOptionError:
-            if isinstance(ce.default, attr.Factory):
-                return attr.NOTHING
-            elif not isinstance(ce.default, Raise):
-                return ce.default
-            raise MissingSecretError(var)
+            return _get_default_secret(var, ce.default)
+
+
+@attr.s
+class DirectorySecrets(object):
+    """Load secrets from a directory containing secrets in separate files.
+    Suitable for reading Docker or Kubernetes secrets
+    from the filesystem inside a container.
+
+    .. versionadded:: 21.1.0
+    """
+
+    secrets_dir = attr.ib()
+    _env_name = attr.ib(default=None)
+
+    @classmethod
+    def from_docker(cls):
```

Comment:
I think this is unnecessarily ambiguous. There's no real benefit over simply writing `from_path("/run/secrets/")`. It might save some typing, but is harder to read so I would remove that helper.